docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+```
+
+##### `ModelSection.tsx`
+
+```jsx
+import React, {useState} from "react";
+
+import PropTypes from 'prop-types';
+import { Tabs, Tab, AppBar, Typography, Paper, Container, useTheme } from '@alteryx/ui';
+
+import { ModelEvaluation, ModelTraining } from "./model-views";
+
+function TabContainer({ children, dir }) {
+  return (
+    <Typography component="div" dir={dir} style={{ padding: 8 * 3 }}>
+      {children}
+    </Typography>
+  );
+}
+
+TabContainer.propTypes = {
+  children: PropTypes.node.isRequired,
+  dir: PropTypes.string
+};
+
+TabContainer.defaultProps = { dir: 'ltr' };
+
+
+const ModelSection = () => {
+ 
+  const theme = useTheme();
+
+  const [selectedTab, setValue] = useState(0);
+
+  const modelTabs = {
+    0: <TabContainer dir={theme.direction}><ModelTraining  /></TabContainer>,
+    1: <TabContainer dir={theme.direction}><ModelEvaluation /></TabContainer>,
+  }
+    
+  return (
+    <Container>
+      <Paper>
+        <AppBar color="default" elevation={0} position="static">
+          <Tabs indicatorColor="primary" onChange={(_, newValue) => {setValue(newValue)}} textColor="primary" value={selectedTab} variant="fullWidth">
+            <Tab label="Training & Evaluation" />
+            <Tab label="Prediction" />
+          </Tabs>
+        </AppBar>
+        {modelTabs[selectedTab]}
+      </Paper>
+    </Container>
+  );
+}
+
+export default ModelSection;
+```
+
+#### Component(s): `./src/components/charting/line-plots.tsx`
+
+
+```jsx
+import React, {useState} from "react";
+
+import { AppBar, Container, useTheme } from '@alteryx/ui';
+
+import {
+  Chart as ChartJS,
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Legend,
+} from 'chart.js';
+import { Line } from 'react-chartjs-2';
+
+ChartJS.register(
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Legend
+);
+
+export const InteractiveLinePlot = ({labels, plotTitle, datasets, ...rest}) => {
+    const theme = useTheme();
+    let options = {
+  responsive: true,
+  plugins: {
+    legend: {
+      position: 'top' as const,
+    },
+    title: {
+      display: true,
+      text: plotTitle,
+        },
+    },
+    };
+
+    const data = {
+        labels,
+        datasets,
+        rest
+    }
+
+    return (
+    <Container>
+        <AppBar color="default" elevation={0} position="static">
+            {plotTitle} <br />
+            <Line options={options} data={data}  />;
+        </AppBar>
+        
+    </Container>
+  );
+}
+```
+
+## Testing End to End
+Now, you should see a GUI that looks like the below (E-Note: will add screencap doing one more fresh build for screens since updates):
+
+
+#### Figure: Tool GUI
+
+[figure here]
+
+
+## Package into a YXI
+
+First, we should ensure we have all our dependencies for the front and backend. Then, we run `create-yxi` to package each of these into a deployable using their respective package managers.
```

Comment:
TW review:

First, we should ensure we have all the dependencies for the front and back ends. Then, we run `create-yxi` to package each of these into a deployable using their respective package managers.